src/main/java/seedu/address/model/person/Group.java

```diff
@@ -35,6 +35,16 @@ public boolean isSameName(Group group) {
         return name.equals(group.name);
     }
 
+    /**
+     * Creates a copy of the current group, to prevent the group from being mutated.
+     * @return a copy of the current Group, with the same name and users.
+     */
+    public Group copy() {
+        Group newGroup = new Group(this.name);
+        newGroup.setPersons(this.asUnmodifiableObservableList());
+        return newGroup;
```

Comment:
Can consider doing this in the form of a copy constructor instead where you pass in a `Group` in the constructor.